src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -903,16 +939,26 @@ Namespace Microsoft.CodeAnalysis.Operations
             Dim type As ITypeSymbol = boundRValuePlaceholder.Type
             Dim constantValue As [Optional](Of Object) = ConvertToOptional(boundRValuePlaceholder.ConstantValueOpt)
             Dim isImplicit As Boolean = boundRValuePlaceholder.WasCompilerGenerated
-            If syntax.IsKind(SyntaxKind.ConditionalAccessExpression) Then
-                ' BoundConditionalAccessReceiver isn't actually used until local rewriting, until then that node will be a
-                ' BoundRValuePlaceholder with a syntax node of the entire conditional access. So we dig through the syntax
-                ' to get the expression being conditionally accessed, and return an IConditionalAccessInstanceOperation
-                ' instead of a PlaceholderOperation
-                syntax = DirectCast(syntax, ConditionalAccessExpressionSyntax).Expression
-                Return New ConditionalAccessInstanceExpression(_semanticModel, syntax, type, constantValue, isImplicit)
-            Else
-                Return New PlaceholderExpression(_semanticModel, syntax, type, constantValue, isImplicit)
+
+            Dim knownParent As BoundNode = TryGetParent(boundRValuePlaceholder)
+            Dim placeholderKind As PlaceholderKind = PlaceholderKind.Unspecified
+
+            If knownParent IsNot Nothing Then
+                Select Case knownParent.Kind
+                    Case BoundKind.ConditionalAccess
+                        ' BoundConditionalAccessReceiver isn't actually used until local rewriting, until then that node will be a
+                        ' BoundRValuePlaceholder with a syntax node of the entire conditional access. So we dig through the syntax
+                        ' to get the expression being conditionally accessed, and return an IConditionalAccessInstanceOperation
+                        ' instead of a PlaceholderOperation
+                        syntax = If(TryCast(syntax, ConditionalAccessExpressionSyntax)?.Expression, syntax)
```

Comment:
>TryCast [](start = 36, length = 7)

Can you point me at the test where this cast fails?

Reply:
> Can you point me at the test where this cast fails?

I don't have a test like this. Do you see any problem with the logic?

Reply:
My only problem is that where before we had a DirectCast, we only have a TryCast now. I didn't see any tests with changed Syntax based on this, so I'm wondering why the change.

---
In reply to: [184238556](https://github.com/dotnet/roslyn/pull/26395#discussion_r184238556) [](ancestors = 184238556)

Reply:
I think it is much better to avoid making decisions based on syntax nodes associated with bound nodes. But it is Ok to try to peek a "better" syntax for an IOperation node when we can, and it doesn't make sense to crash when we cannot.

Reply:
> My only problem is that where before we had a DirectCast, we only have a TryCast now. I didn't see any tests with changed Syntax based on this, so I'm wondering why the change.

The old code was checking syntax Kind before doing DirectCast, so the cast was safe. The new code doesn't check the syntax kind first.

---
In reply to: [184238987](https://github.com/dotnet/roslyn/pull/26395#discussion_r184238987) [](ancestors = 184238987,184238556)